src/dbg/expressionfunctions.h

```diff
@@ -5,20 +5,26 @@
 class ExpressionFunctions
 {
 public:
-    using CBEXPRESSIONFUNCTION = std::function<duint(int argc, const duint* argv)>;
+    using CBEXPRESSIONFUNCTION             = std::function<duint(int argc, const duint* argv)>;
+    using CBEXPRESSIONFUNCTIONWITHUSERDATA = std::function<duint(int argc, const duint* argv, void* user)>;
 
     static void Init();
     static bool Register(const String & name, int argc, CBEXPRESSIONFUNCTION cbFunction);
+    static bool Register(const String & name, int argc, CBEXPRESSIONFUNCTIONWITHUSERDATA cbFunction, void* user);
     static bool Unregister(const String & name);
     static bool Call(const String & name, const std::vector<duint> & argv, duint & result);
     static bool GetArgc(const String & name, int & argc);
 
 private:
     struct Function
     {
+        CBEXPRESSIONFUNCTION cbFunction;

```

Comment:
I'm not terribly happy with this implementation, but I figured the options were: 

- Make everything require the userdata invocation. This would break existing uses of Register, and in doing so making call sites that didn't need a user data pointer to be uglier
- Make Function an interface and have two subclasses. Downside here is we'd have to change the map to point to a pointer of that interface; and manage some dynamic allocations. 
- Tagged unions for the CB function; however VS2013 doesn't let you do C++11 union stuff and C++03 doesn't let you do unions on complex types like std::function. 
- Wrap functions without a user data arg in a function that takes in the CBEXPRESSIONFUNCTION as it's user data function. Seemed unnecessarily maze like though; and I'm not actually sure if you could find a good way to cast CBEXPRESSIONFUNCTION to void*.  

In the end I just tried to keep it simple and easy to follow through the code.